lib/socketry/exceptions.rb

```diff
@@ -0,0 +1,18 @@
+module Socketry
+  # Generic catch all for all Socketry errors
+  SocketError = Class.new(IOError)
```

Comment:
Mmm... I can see how `X = Class.new(Y)` might look better than `class X < Y; end` but I think it makes code less consistent :D Anyway, I have no strong feeling one way or the other ;))